packages/expo-updates/ios/EXUpdates/UpdatesModule.swift

```diff
@@ -21,6 +21,8 @@ public final class UpdatesModule: Module {
 
   public required init(appContext: AppContext) {
     updatesService = appContext.legacyModule(implementing: EXUpdatesModuleInterface.self)
+    // Ensures the universal UpdatesConfig can cast to versioned UpdatesConfig without exception in Swift
+    object_setClass(updatesService?.config, UpdatesConfig.self)
```

Comment:
we will have versioned ABI49_0_0UpdatesModules and ABI49_0_0UpdatesServices. however, the UpdatesConfig is from unversioned code. swift will assume the `updatesService?.config` here would be ABI49_0_0UpdatesConfig, but it is actually an unversioned `UpdatesConfig`. swift will then throw EXC_BAD_ACCESS runtime crash because of wrong type. i think it is checking the `ISA` pointer. we have `object_setClass` here to change the ISA pointer to the unversioned UpdatesConfig.